tutorial/unity_local_realm.md

```diff
@@ -0,0 +1,608 @@
+
+In the following section I will explain how to make sure that the current game state gets saved and the players can resume the game at any state.
+
+## <a name="add-realm"></a>Adding Realm to your project
+
+The first thing we need to do is to import the Realm framework into Unity.
+The easiest way to do this is by using NPM.
+
+You'll find it via `Windows` → `Package Manager` → cogwheel in the top right corner → `Advanced Project Settings`:
+
+![Package Manager](images/03_package_manager.jpg)
+
+Within the `Scoped Registries` you can add the `Name`, `URL` and `Scope` as follows:
+
+![Project Settings](images/04_project_settings.jpg)
+
+This adds `NPM` as a source for libraries. The final step is to tell the project which dependencies to actually integrate into the project. This is done in the `manifest.json` file which is located in the `Packages` folder of your project.
+
+Here you need to add the following line to the `dependencies`:
+
+```json
+"io.realm.unity": "<version-number>"
+```
+
+Replace `<version-number>` with the most recent Realm version found in https://github.com/realm/realm-dotnet/releases and you're all set.
+
+The final `manifest.json` should look something like this:
+
+```json
+{
+ "dependencies": {
+   ...
+   "io.realm.unity": "10.3.0"
+ },
+ "scopedRegistries": [
+   {
+     "name": "NPM",
+     "url": "https://registry.npmjs.org/",
+     "scopes": [
+       "io.realm.unity"
+     ]
+   }
+ ]
+}
+```
+
+When you switch back to Unity it will reload the dependencies. If you then open the `Package Manager` again, you should see `Realm` as a new entry in the list on the left:
+
+![Realm in Project Manager](images/05_realm_in_project_manager.jpg)
+
+We can now start using Realm in our Unity project.
+
+## Top-down or bottom-up?
+
+Before we actually start adding Realm to our code we need to think about how we want to achieve this and how the UI and database will interact with each other.
+
+There are basically two options we can choose from: top-down or bottom-up.
+
+The top-down approach would be to have the UI drive the changes. The `Piece` would know about it's database object and whenever a `Piece` is moved it would also update the database with it's new position.
+
+The preferred approach would be bottom-up though. Changes will be applied to the Realm and it will then take care of whatever implications this has on the UI by sending notifications.
+
+Let's first look into the initial setup of the board.
+
+## Setting up the board
+
+The first thing we want to do is to define a Realm representation of our Piece since we cannot save the `MonoBehaviour` directly in Realm. Classes that are supposed to be saved in Realm need to subclass `RealmObject`. The class `PieceEntity` will represent such an object. Note that we can not just duplicate the types from `Piece` since [not all of them can be saved in Realm](https://docs.mongodb.com/realm/sdk/dotnet/data-types/field-types/), like `Vector3` and `enum`.
```

Comment:
```suggestion
The first thing we want to do is to define a Realm representation of our Piece since we cannot save the `MonoBehaviour` directly in Realm. Classes that are supposed to be saved in Realm need to subclass `RealmObject`. The class `PieceEntity` will represent such an object. Note that we cannot just duplicate the types from `Piece` since [not all of them can be saved in Realm](https://docs.mongodb.com/realm/sdk/dotnet/data-types/field-types/), like `Vector3` and `enum`.
```